Assets/BossRoom/Scripts/Client/UI/HeroActionBar.cs

```diff
@@ -128,10 +113,62 @@ void OnButtonClickedUp(int buttonIndex)
 
             switch (buttonIndex)
             {
-                case 0: m_InputSender.RequestAction(m_CharacterData.Skill1, SkillTriggerStyle.UIRelease); break;
-                case 1: m_InputSender.RequestAction(m_CharacterData.Skill2, SkillTriggerStyle.UIRelease); break;
-                case 2: m_InputSender.RequestAction(m_CharacterData.Skill3, SkillTriggerStyle.UIRelease); break;
+                case 0: m_InputSender.RequestAction(m_IsOtherPlayerSelected ? ActionType.GeneralRevive : m_NetState.CharacterData.Skill1, SkillTriggerStyle.UIRelease); break;
+                case 1: m_InputSender.RequestAction(m_NetState.CharacterData.Skill2, SkillTriggerStyle.UIRelease); break;
+                case 2: m_InputSender.RequestAction(m_NetState.CharacterData.Skill3, SkillTriggerStyle.UIRelease); break;
+            }
+        }
+
+        void UpdateAllIcons()
+        {
+            UpdateIcon(0, m_NetState.CharacterData.Skill1);
+            UpdateIcon(1, m_NetState.CharacterData.Skill2);
+            UpdateIcon(2, m_NetState.CharacterData.Skill3);
+
+            if (m_NetState.TargetId.Value != 0
+                && NetworkSpawnManager.SpawnedObjects.TryGetValue(m_NetState.TargetId.Value, out NetworkObject selection)
+                && selection != null
+                && selection.IsPlayerObject
+                && selection.NetworkObjectId != m_NetState.NetworkObjectId)
+            {
+                m_IsOtherPlayerSelected = true;
+                // we have a different player selected! In that case we want to reflect that our basic Action is a Revive, not an attack
+                UpdateIcon(0, ActionType.GeneralRevive);
```

Comment:
This could be added now or in a new PR, but I find that we should have a disabled sprite for revival when the target is a player that is still alive. The button should only turn into the active revive action when they're truly actually fainted.

Reply:
let's do this in this PR

Reply:
I've gotten two slightly different instructions on this one: the notes here are to disable the "revive" button when the selected player isn't revivable, but out of band, it sounds like maybe I'm supposed to "hide" the revive button, presumably showing the basic-attack icon if the selected character is alive. I've implemented the "disable" approach, but can easily switch it just showing a non-disabled basic-attack icon instead.

Note that as part of this change we need to check the selected player's NetState during `Update()` to see if their aliveness state has changed. To implement this efficiently we cache the selected player's NetState.

ok (but please ensure this is the intended behavior)

Reply:
I don't object to this implementation. A disabled hit action might be the more "correct" implementation, but I do see the argument for having it like this. It subtly shows the user that you "can" revive another player, just not now. I'll let others weigh in.